saleor/core/search_tasks.py

```diff
@@ -104,3 +109,32 @@ def set_search_document_values(
         f"{progress}% done."
     )
     return updated_count
+
+
+def set_search_vector_values(
+    qs,
+    total_count,
+    updated_count,
+    prepare_search_document_func,
+    prepare_search_vector_func,
+):
+    Model = qs.model
+    instances = []
+    for instance in qs:
+        instance.search_document = prepare_search_document_func(
+            instance, already_prefetched=True
+        )
+        instance.search_vector = prepare_search_vector_func(
+            instance, already_prefetched=True
+        )
+        instances.append(instance)
+    Model.objects.bulk_update(instances, ["search_document", "search_vector"])
+
+    updated_count += len(instances)
+    progress = round((updated_count / total_count) * 100, 2)
+
+    task_logger.info(
+        f"Updated {updated_count} from {total_count} {Model.__name__.lower()}s - "
+        f"{progress}% done."
```

Comment:
Shouldn't we pass values as params?

Reply:
This is not new code, I created a vector version of all search_document functions so we can compare them while this is in draft.

Reply:
> In #8734 we drop fixes from #8158 about cursors builds from ranks. Should we revert it?

Good catch, we should.